tradeit-android-sdk/src/main/java/it/trade/android/sdk/TradeItSdk2.java

```diff
@@ -0,0 +1,64 @@
+package it.trade.android.sdk;
+
+import android.content.Context;
+import android.support.annotation.WorkerThread;
+
+import it.trade.android.sdk.exceptions.TradeItSDKConfigurationException;
+import it.trade.android.sdk.manager.TradeItLinkedBrokerManager;
+import it.trade.android.sdk.model.TradeItLinkedBrokerCache;
+import it.trade.tradeitapi.exception.TradeItKeystoreServiceCreateKeyException;
+import it.trade.tradeitapi.exception.TradeItRetrieveLinkedLoginException;
+import it.trade.tradeitapi.model.TradeItEnvironment;
+
+/**
+ * Created by sanbeg on 3/29/17.
+ */
+
+public class TradeItSdk2 {
+    private final String apiKey;
+    private final TradeItEnvironment environment;
+    private TradeItLinkedBrokerManager linkedBrokerManager;
+    private final TradeItLinkedBrokerCache linkedBrokerCache;
+
+    public TradeItSdk2(Context context, String apiKey, TradeItEnvironment environment) {
+        this.apiKey = apiKey;
+        this.environment = environment;
+        linkedBrokerCache = new TradeItLinkedBrokerCache(context);
+    }
+
+    @WorkerThread
+    public void configure(Context context) {
+        try {
+            // On a fresh install, this could take multiple seconds to generate a key pair
```

Comment:
It should be OK then.  That would definitely mean we'd have to construct the object on a background thread, but we probably would've needed to that regardless, so it may be better to make everything final and initialize it all in the constructor.